### PR TITLE
Add exercise: resistor-color

### PR DESCRIPTION
**Pull request by IsaacG**

### CODE REVIEW

File: exercises/practice/resistor-color/.meta/config.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+  "authors": [],
```
Comment (BNAndras) [inline exercises/practice/resistor-color/.meta/config.json:2]: Username is missing here.

Comment (IsaacG) [reply exercises/practice/resistor-color/.meta/config.json:2]: No anonymous exercises?

Corrected! Thanks for the catch.

File: exercises/practice/resistor-color/resistor-color_test.sql [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+-- Update message for failed tests to give helpful information:
+UPDATE test_color_code
+SET message = 'Result for ' || l.color || ' is "' || l.result || '", but should be: "' || test_color_code.result || '"'
+FROM (SELECT color, result FROM color_code) AS l
+WHERE l.color = test_color_code.color AND test_color_code.status = 'fail';
+
+-- Comparison of user input and the tests updates the status for each test:
+UPDATE test_colors
+SET status = 'pass'
+FROM (SELECT GROUP_CONCAT("color" ORDER BY "index") as "result" FROM "colors") AS l
```
Comment (glennj) [inline exercises/practice/resistor-color/resistor-color_test.sql:27]: No "order by" in sqlite group_concat: https://sqlite.org/lang_aggfunc.html#group_concat

File: exercises/practice/resistor-color/resistor-color_test.sql [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+WHERE l.result = test_colors.result;
+-- Update message for failed tests to give helpful information:
+UPDATE test_colors
+SET message = 'Result for Colors is "' || l.result || '", but should be: "' || test_colors.result || '"'
+FROM (SELECT GROUP_CONCAT("color" ORDER BY "index") as "result" FROM "colors") AS l
```
Comment (glennj) [inline exercises/practice/resistor-color/resistor-color_test.sql:32]: no "order by"

File: exercises/practice/resistor-color/resistor-color.sql [hunk 4]
```diff
@@ -0,0 +1,5 @@
+-- Task one: Populate "colors" table with colors and index/value.
+-- Schema: CREATE TABLE "colors" ("index" INT, "color" TEXT);
+--
+-- Task two: Update "color_code" with the correct value/result based on the color.
+-- Schema: CREATE TABLE "color_code" ("color" TEXT, "result" INT);
```
Comment (glennj) [inline exercises/practice/resistor-color/resistor-color.sql:5]: The descriptions of the tasks would be better (or, additionally) placed in `.docs/instructions.append.md`.

File: exercises/practice/resistor-color/resistor-color_test.sql (hunk 3 above)
Comment (glennj) [inline exercises/practice/resistor-color/resistor-color_test.sql:32]: Can we get away from using `l` as a table alias please? It's hard to read and meaningless.

File: exercises/practice/resistor-color/resistor-color_test.sql (hunk 2 above)
Comment (glennj) [inline exercises/practice/resistor-color/resistor-color_test.sql:27]: Can we structure this exercise in such a way that the student has to come up with that sub-select by themselves? Perhaps we give them the `colors` table complete with index values, and demand they populate a column in some other result table with the comma separated colours in the right order.

**Review by glennj — CHANGES_REQUESTED** [review]
Thanks @IsaacG

File: exercises/practice/resistor-color/resistor-color_test.sql (hunk 2 above)
Comment (IsaacG) [reply exercises/practice/resistor-color/resistor-color_test.sql:27]:
```
sqlite> SELECT GROUP_CONCAT("color" ORDER BY "index") AS "result" FROM "colors";
+------------------------------------------------------------+
|                           result                           |
+------------------------------------------------------------+
| black,brown,red,orange,yellow,green,blue,violet,grey,white |
+------------------------------------------------------------+

sqlite> SELECT GROUP_CONCAT("color") AS "result" FROM "colors";
+------------------------------------------------------------+
|                           result                           |
+------------------------------------------------------------+
| blue,orange,violet,red,brown,black,white,green,yellow,grey |
+------------------------------------------------------------+

File: exercises/practice/resistor-color/resistor-color.sql (hunk 4 above)
Comment (IsaacG) [reply exercises/practice/resistor-color/resistor-color.sql:5]: Oh. Yeah. I/We should probably utilize those :D We may need them heavily for this track. Will add one!

File: exercises/practice/resistor-color/resistor-color_test.sql (hunk 3 above)
Comment (IsaacG) [reply exercises/practice/resistor-color/resistor-color_test.sql:32]: Absolutely. I don't like single-char variables.

File: exercises/practice/resistor-color/resistor-color_test.sql (hunk 2 above)
Comment (IsaacG) [reply exercises/practice/resistor-color/resistor-color_test.sql:27]: I'm a bit hesitant to wander too far from the problem spec here.

Would that be testing that they can use a `GROUP_CONCAT`? Is that related to the problem spec? Is that more of a concept exercise?

Should we just skip this test?

File: exercises/practice/resistor-color/.docs/instructions.append.md [hunk 5]
```diff
@@ -0,0 +1,15 @@
+# SQLite specific instructions
+
+This exercise has two parts.
+
+The first part requires that you popular a `colors` table with all ten colors.
```
Comment (vaeng) [inline exercises/practice/resistor-color/.docs/instructions.append.md:5]:
```suggestion
The first part requires that you populate a `colors` table with all ten colors.
```

**Comment by IsaacG** [comment]
Ping?

**Review by glennj — APPROVED** [review]

File: exercises/practice/resistor-color/.docs/instructions.append.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (IsaacG) [reply exercises/practice/resistor-color/.docs/instructions.append.md]: Fixed!

File: exercises/practice/resistor-color/resistor-color_test.sql (hunk 2 above)
Comment (IsaacG) [reply exercises/practice/resistor-color/resistor-color_test.sql]: I'm not sure what version I'm running that does this but the CI doesn't like it. I'll see if I can find another way to make this test work.

**Comment by IsaacG** [comment]
@glennj @vaeng @BNAndras How do you feel about just dropping the "insert all colors" part of this exercise and sticking with a "update values for colors" bit?

**Comment by BNAndras** [comment]
Personally, I never did care for listing out colors because it’s not carried over to the other resistor-color exercises so I wouldn’t miss it.

**Comment by glennj** [comment]
I get the feeling that we're going to need some more less-difficult exercises here. :+1: from me.

File: exercises/practice/resistor-color/resistor-color_test.sql (hunk 3 above)
Comment (IsaacG) [reply exercises/practice/resistor-color/resistor-color_test.sql]: Dropped that test and table.

File: exercises/practice/resistor-color/resistor-color_test.sql (hunk 2 above)
Comment (IsaacG) [reply exercises/practice/resistor-color/resistor-color_test.sql]: Skipped.

**Review by BNAndras — APPROVED** [review]

**Review by vaeng — APPROVED** [review]

**Pull request merged**